plutus-scb/src/Plutus/SCB/ParseStringifiedJSON.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia        #-}
+{-# LANGUAGE FlexibleContexts   #-}
+{-# LANGUAGE LambdaCase         #-}
+{-# LANGUAGE MonoLocalBinds     #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE TypeApplications   #-}
+module Plutus.SCB.ParseStringifiedJSON(
+    UnStringifyJSONLog(..),
+    parseStringifiedJSON
+    ) where
+
+import           Cardano.BM.Data.Tracer        (ToObject)
+import           Cardano.BM.Data.Tracer.Extras (PrettyToObject (..))
+import           Control.Monad.Freer
+import           Control.Monad.Freer.Log       (LogMsg, logDebug)
+import           Data.Aeson                    (FromJSON, ToJSON, Value (..), decode)
+import qualified Data.ByteString.Lazy.Char8    as LBS
+import qualified Data.Text.Encoding            as Text
+import           Data.Text.Prettyprint.Doc     (Pretty (..))
+import           GHC.Generics                  (Generic)
+
+data UnStringifyJSONLog =
+    ParseStringifiedJSONAttempt
+    | ParseStringifiedJSONFailed
+    | ParseStringifiedJSONSuccess
+    deriving stock (Eq, Show, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+    deriving ToObject via (PrettyToObject UnStringifyJSONLog)
+
+instance Pretty UnStringifyJSONLog where
+    pretty = \case
+        ParseStringifiedJSONAttempt -> "parseStringifiedJSON: Attempting to remove 1 layer StringifyJSON"
+        ParseStringifiedJSONFailed -> "parseStringifiedJSON: Failed, returning original string"
+        ParseStringifiedJSONSuccess -> "parseStringifiedJSON: Succeeded"
+
+parseStringifiedJSON ::
```

Comment:
Why do we actually have this?

Reply:
Some of the values that we get from the frontend contain string-encoded JSON values... (at least that was the case before the demo at the virtual summit, when I first added `parseStringifiedJSON`. I'll check if it is still needed)